modules/Bio/EnsEMBL/Hive/RunnableDB/FastaFactory.pm

```diff
@@ -130,26 +131,35 @@ sub write_output {
 
     my $input_seqio         = $self->param('input_seqio');
     my $max_chunk_length    = $self->param('max_chunk_length');
+    my $output_dir          = $self->param('output_dir');
     my $output_prefix       = $self->param('output_prefix');
     my $output_suffix       = $self->param('output_suffix');
 
     my $chunk_number = 1;   # counts the chunks
     my $chunk_length = 0;   # total length of the current chunk
     my $chunk_size   = 0;   # number of sequences in the current chunk
-    my $chunk_name   = $output_prefix.$chunk_number.$output_suffix;
+    
+    my $current_output_directory;    
     if ($self->param('hash_directories')) {
         my $dir_tree = dir_revhash($chunk_number);
-        if ($dir_tree ne '') {
-            mkpath($dir_tree);
-            $chunk_name = $dir_tree.'/'.$chunk_name;
+        
+        if ($dir_tree eq '') {
+	  $current_output_directory = $output_dir;
+        } else {
+	  $current_output_directory = $output_dir . '/' . $dir_tree;
         }
+    } else {      
+      $current_output_directory = $output_dir;
     }
-    my $chunk_seqio  = Bio::SeqIO->new(-file => '>'.$chunk_name, -format => 'fasta');
+    mkpath($current_output_directory);
+    my $chunk_name   = $current_output_directory . '/' . $output_prefix . $chunk_number . $output_suffix;
+    my $chunk_seqio  = Bio::SeqIO->new(-file => '>' . $chunk_name, -format => 'fasta');
     
-    while (my $seq_object = $input_seqio->next_seq) {
-	$chunk_seqio->write_seq( $seq_object );
+    while (my $seq_object = $input_seqio->next_seq) {	
 	
         if((my $seq_length = $seq_object->length()) + $chunk_length <= $max_chunk_length) {
+        
+	    $chunk_seqio->write_seq( $seq_object );
```

Comment:
The reason why I moved the write_seq statement was that in the original version $chunk_length and $chunk_size were wrong, because the last sequence written to the file, but $chunk_length and $chunk_size weren't updated to account for that.

Point taken about the empty files. Even worse, my version was omitting the last sequence. I'm fixing this.